File: src/main/java/me/maxih/aquaticcommandstestplugin/PingCommand.java
package me.maxih.aquaticcommandstestplugin;

import me.maxih.aquaticcommands.framework.Command;
import me.maxih.aquaticcommands.framework.CommandException;
import me.maxih.aquaticcommands.framework.LiteralNode;
import me.maxih.aquaticcommands.framework.util.ParsedArgument;
import org.bukkit.command.CommandSender;
import scala.collection.Seq;
import scala.collection.immutable.Map;

public class PingCommand implements Command {
    private LiteralNode rootNode = new LiteralNode("ping", true);

    @Override
    public LiteralNode rootNode() {
        return rootNode;
    }

    @Override
    public void execute(CommandSender sender, Map<String, ParsedArgument> arguments, Seq<String> literals) throws CommandException {
        sender.sendMessage("Pong!");
    }

}

File: src/main/scala/me/maxih/aquaticcommandstestplugin/PingCommand.scala
package me.maxih.aquaticcommandstestplugin

import me.maxih.aquaticcommands.framework.util.ParsedArgument
import me.maxih.aquaticcommands.framework.{Command, LiteralNode}
import org.bukkit.command.CommandSender

/**
  * Created by Maxi H. on 13.09.2018
  */
class PingCommand extends Command {
  override val rootNode: LiteralNode = new LiteralNode("ping", true)

  override def execute(sender: CommandSender, arguments: Map[String, ParsedArgument], literals: Seq[String]): Unit =
    sender.sendMessage("Pong!")
}
